lib/core.ps1

```diff
@@ -857,15 +857,38 @@ function ensure_in_path($dir, $global) {
     }
 }
 
-function ensure_architecture($architecture_opt) {
-    if(!$architecture_opt) {
-        return default_architecture
+function Get-DefaultArchitecture {
+    $arch = get_config DEFAULT_ARCHITECTURE
+    $system = if (${env:ProgramFiles(Arm)}) {
```

Comment:
Couldn't this also mean `arm32`? We should check for Is64BitOperatingSystem first. 

(Windows has arm32 since 8.1, I think).

Reply:
This is for all ARM-supported systems (only Windows 10 and Windows 11, Windows 8.1 doesn't support x86 apps even), and it contains 32-bit ARM apps.

So I think if there is `${env:ProgramFiles(Arm)}` in scoop's use case, it is Windows 10 or Windows 11 and it is ARM64, or we have nothing to install for them.